File: hexample-parent/domain/src/main/java/at/willhaben/sp/hexample/domain/jobsearch/services/JobSearchService.java
package at.willhaben.sp.hexample.domain.jobsearch.services;

import at.willhaben.sp.hexample.domain.jobsearch.ports.JobSearchDatabasePort;
import at.willhaben.sp.hexample.domain.jobsearch.ports.JobSearchServicePort;
import at.willhaben.sp.hexample.domain.model.JobOffer;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class JobSearchService implements JobSearchServicePort {

    private JobSearchDatabasePort jobSearchDatabase;

    public JobSearchService(JobSearchDatabasePort jobSearchDatabase) {
        this.jobSearchDatabase = jobSearchDatabase;
    }

    @Override
    public List<JobOffer> searchForJobsWithTitle(String title) {
        return jobSearchDatabase.searchForJobsWithTitle(title);
    }

}

File: hexample-parent/domain/src/main/kotlin/at/willhaben/sp/hexample/domain/jobsearch/services/JobSearchService.kt
package at.willhaben.sp.hexample.domain.jobsearch.services

import at.willhaben.sp.hexample.domain.jobsearch.ports.JobSearchDatabasePort
import at.willhaben.sp.hexample.domain.jobsearch.ports.JobSearchServicePort
import at.willhaben.sp.hexample.domain.model.JobOffer
import org.springframework.stereotype.Service

@Service
class JobSearchService(
    private val jobSearchDatabasePort: JobSearchDatabasePort
) : JobSearchServicePort {
    override fun searchForJobsWithTitle(title: String?): List<JobOffer> {
        return jobSearchDatabasePort.searchForJobsWithTitle(title)
    }
}
